manager/controllers/app/blueprint_controller.go

```diff
@@ -163,29 +168,129 @@ func (r *BlueprintReconciler) reconcile(ctx context.Context, log logr.Logger, bl
 		// Add metadata arguments (namespace, name, labels, etc.)
 		args["metadata"] = map[string]interface{}{
 			"name":      utils.Hash(step.Name, 20),
-			"namespace": blueprint.Namespace, // TODO(roee.shlomo): should eventually be a dedicated namespace per M4DApplication
+			"namespace": blueprint.Namespace,
 			"labels":    blueprint.Labels,
 		}
+		releaseName := "r" + args["metadata"].(map[string]interface{})["name"].(string)
```

Comment:
There is a release for each Helm chart - a Helm concept.

Reply:
why are you concatenation in "r" the chart release is typically a unique  string concatenated to the chart name

Reply:
> 
> 
> why are you concatenation in "r" the chart release is typically a unique string concatenated to the chart name

I did not change the original release naming, just moved this line around. I believe this was your comment (in applyChartResource function):
`// we add the "r" character at the beginning of the release name, since it must begin with an alphabetic character`

Reply:
can you point me to the original code which had this "r"?

Reply:
It was added by @cdoron to ensure that the release name starts with a legal character (a numeric is invalid for the first char)